bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/connectivity/rawdata/LocalConnectRawDataBean.java

```diff
@@ -0,0 +1,245 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solax.internal.connectivity.rawdata;
+
+import java.util.Arrays;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.solax.internal.model.InverterData;
+import org.openhab.binding.solax.internal.model.InverterType;
+import org.openhab.binding.solax.internal.util.GsonSupplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link LocalConnectRawDataBean} collects the raw data and the specific implementation to return the parsed data.
+ * If there are differences between the inverters probably would be wise to split the parsing in seprate class(es)
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class LocalConnectRawDataBean implements RawDataBean, InverterData {
+
+    private final Logger logger = LoggerFactory.getLogger(LocalConnectRawDataBean.class);
+
+    private @Nullable String sn;
+    private @Nullable String ver;
+    private int type;
+    private short @Nullable [] Data;
+    private String @Nullable [] Information;
+    private @Nullable String rawData;
+
+    @Override
+    public String toString() {
+        return "LocalConnectRawDataBean [sn=" + sn + ", ver=" + ver + ", type=" + type + ", Information="
+                + Arrays.toString(Information) + ", Data=" + Arrays.toString(Data) + "]";
+    }
+
+    public @Nullable String getSn() {
+        return sn;
+    }
+
+    public void setSn(@Nullable String sn) {
+        this.sn = sn;
+    }
+
+    public @Nullable String getVer() {
+        return ver;
+    }
+
+    public void setVer(@Nullable String ver) {
+        this.ver = ver;
+    }
+
+    public int getType() {
+        return type;
+    }
+
+    public void setType(int type) {
+        this.type = type;
+    }
+
+    public short @Nullable [] getData() {
+        return Data;
+    }
+
+    public void setData(short @Nullable [] data) {
+        Data = data;
+    }
+
+    public String @Nullable [] getInformation() {
+        return Information;
+    }
+
+    public void setInformation(String @Nullable [] information) {
+        Information = information;
+    }
+
+    @Override
+    public @Nullable String getRawData() {
+        return rawData;
+    }
+
+    public void setRawData(String rawData) {
+        this.rawData = rawData;
+    }
+
+    public static LocalConnectRawDataBean fromJson(String json) {
+        Gson gson = GsonSupplier.getInstance();
+        LocalConnectRawDataBean deserializedObject = gson.fromJson(json, LocalConnectRawDataBean.class);
+        deserializedObject.setRawData(json);
```

Comment:
You can fix the warning _"Potential null pointer access: The variable deserializedObject may be null at this location"_ here by checking for null and throwing an exception. If you make sure that provided String `json` is neither null or empty, then it should not happen. So you could even throw `IllegalStateException`, since this is a runtime exception you won't have to handle (however, it will crash the binding if it does happen).

Reply:
I think that's going to be OK to throw such an exception. I do not intend usage of this method with a nullable value. If you use it with a null (for some reason), you should fix it immediately, otherwise there is no data.... :)

Reply:
Weird thing is though... that when I make an explicit null check on the argument (or the deserializedObject) Eclipse says "dead code" or "redundant null check". Probably because the json argument is NonNullByDefault. I'm really not very experienced with these annotations, so would appreciate any help here...

Reply:
👍 When you receive the payload, you just need to verify that you don't call this method with an empty String, since this would lead to the same situation.

Reply:
If I make: 
` if (json == null || json.isEmpty()) {
            throw new IllegalStateException("Argument should be not null or empty.");
        }`
We get now two warnings 🤣 
`[WARNING] E:\git\openhab-addons\bundles\org.openhab.binding.solax\src\main\java\org\openhab\binding\solax\internal\connectivity\rawdata\LocalConnectRawDataBean.java:[104,13] Redundant null check: comparing '@NonNull String' against null
[WARNING] E:\git\openhab-addons\bundles\org.openhab.binding.solax\src\main\java\org\openhab\binding\solax\internal\connectivity\rawdata\LocalConnectRawDataBean.java:[109,9] Potential null pointer access: The variable deserializedObject may be null at this location
`

Reply:
If I make it like this:
`  if (json.isEmpty()) {
            throw new IllegalStateException("Argument should be not null or empty.");
        }
        Gson gson = GsonSupplier.getInstance();
        LocalConnectRawDataBean deserializedObject = gson.fromJson(json, LocalConnectRawDataBean.class);
        if (!Objects.isNull(deserializedObject)) {
            deserializedObject.setRawData(json);
        }....`
it gives a compilation error:
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project org.openhab.binding.solax: Compilation failure
[ERROR] E:\git\openhab-addons\bundles\org.openhab.binding.solax\src\main\java\org\openhab\binding\solax\internal\connectivity\rawdata\LocalConnectRawDataBean.java:[113,16] Null type mismatch: required 'org.openhab.binding.solax.internal.connectivity.rawdata.@NonNull LocalConnectRawDataBean' but the provided value is inferred as @org.eclipse.jdt.annotation.Nullable
`

Reply:
There is really something weird here or I don't see the elephant in the room...